source/common/router/retry_state_impl.cc

```diff
@@ -36,7 +37,7 @@ RetryStatePtr RetryStateImpl::create(const RetryPolicy& route_policy,
 
   // We short circuit here and do not both with an allocation if there is no chance we will retry.
   if (request_headers.EnvoyRetryOn() || request_headers.EnvoyRetryGrpcOn() ||
-      route_policy.retryOn()) {
+      route_policy.retryOn() || !route_policy.retriableStatusCodes().empty()) {
```

Comment:
Is it worth adding this conditional here? It might short circuit some computation but seems not worth it to me since I think it will still work correctly if we get into the policy but there are no codes? Up to you.

Reply:
I'll just remove it